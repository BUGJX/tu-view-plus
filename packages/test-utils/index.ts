export { default as defineGetter } from './define-getter';
