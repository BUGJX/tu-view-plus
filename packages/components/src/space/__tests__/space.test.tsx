import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'
import Space from '../src/space.vue'

const AXIOM = 'Tu view is good'

describe('Space', () => {
  test('render test', () => {
    const wrapper = mount(() => <Space>{AXIOM}</Space>)

    expect(wrapper.text()).toEqual(AXIOM)
  })

  test('size', () => {
    const sizes = ['mini', 'small', 'medium', 'large']
    sizes.forEach(size => {
      const wrapper = mount(Space, {
        props: { size },
        slots: {
          default: [`<div>${AXIOM}</div>`, `<div>${AXIOM}</div>`]
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
