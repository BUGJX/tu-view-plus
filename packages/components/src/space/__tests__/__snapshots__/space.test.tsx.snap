// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Space > size 1`] = `
"<div class=\\"tu-space tu-space--horizontal tu-space--align-center\\">
  <!---->
  <div class=\\"tu-space__item\\" style=\\"margin-right: 4px;\\">
    <div>Tu view is good</div>
  </div>
  <!---->
  <div class=\\"tu-space__item\\">
    <div>Tu view is good</div>
  </div>
</div>"
`;

exports[`Space > size 2`] = `
"<div class=\\"tu-space tu-space--horizontal tu-space--align-center\\">
  <!---->
  <div class=\\"tu-space__item\\" style=\\"margin-right: 8px;\\">
    <div>Tu view is good</div>
  </div>
  <!---->
  <div class=\\"tu-space__item\\">
    <div>Tu view is good</div>
  </div>
</div>"
`;

exports[`Space > size 3`] = `
"<div class=\\"tu-space tu-space--horizontal tu-space--align-center\\">
  <!---->
  <div class=\\"tu-space__item\\" style=\\"margin-right: 16px;\\">
    <div>Tu view is good</div>
  </div>
  <!---->
  <div class=\\"tu-space__item\\">
    <div>Tu view is good</div>
  </div>
</div>"
`;

exports[`Space > size 4`] = `
"<div class=\\"tu-space tu-space--horizontal tu-space--align-center\\">
  <!---->
  <div class=\\"tu-space__item\\" style=\\"margin-right: 24px;\\">
    <div>Tu view is good</div>
  </div>
  <!---->
  <div class=\\"tu-space__item\\">
    <div>Tu view is good</div>
  </div>
</div>"
`;
